core/src/main/java/com/datastrato/gravitino/storage/kv/KvGarbageCollector.java

```diff
@@ -232,6 +253,71 @@ private void collectAndRemoveOldVersionData() throws IOException {
         kvBackend.delete(kv.getKey());
       }
     }
+
+    commitIdHasBeenCollected = kvs.isEmpty() ? startKey : kvs.get(0).getKey();
+    kvBackend.put(LAST_COLLECT_COMMIT_ID_KEY, commitIdHasBeenCollected, true);
```

Comment:
What if this `LAST_COLLECT_COMMIT_ID_KEY ` is missing for the first time?

Reply:
I guess you may want to ask what if the value of `LAST_COLLECT_COMMIT_ID_KEY` is null or missing in the first time.

In the first time, if we do not set the value, then it would be null and we will set it to the default one `endOfTransactionId`,  that mean we would start to scan from the minimum transaction id to the latest one. Once we complete the task, we will save the largest transaction ID that has been handled and use it as the starting ID the next time.